web/modules/custom/bglobal_pardot/bglobal_pardot.module

```diff
@@ -83,10 +86,13 @@ function bglobal_pardot_trackers() {
     $page_title = 'Berkeley Global';
   }
 
+  $current_url = Url::fromRoute('<current>', [], ['query' => \Drupal::request()->query->all(), 'absolute' => 'true'])->toString();
   $trackers = [
     'full_path' => 'This value needs to be derived from local storage and set as a script target object.',
     'page_title' => $page_title,
     'program_name' => '',
+    'success_location' => $current_url . "?thankyou=welcome",
```

Comment:
This will require some JS to handle.

Reply:
What part? As I have it set up on my side, I have some javaScript to add '?' if there are no queries present and '&' if there are, I figured we could do similar string parsing on the $current_url variable with PHP, right?

Reply:
To handle the display message on the return request.

Understand that Drupal is a giant front-end controller and is not really set up to handle `?foo=bar` requests. The actual server request for the current page is something like `global.berkeley.edu/?q=international/program`. The Drupal routing system won't know what to do with the `?thankyou` parameter.

The JS (or possibly PHP code) would need to grab that parameter and do something with it.

Reply:
The more Drupal way to handle it is just to have a single Success and Error page that you direct people to at all times, they way 403 and 404 pages are handled.

Reply:
Oh, that makes sense. Yeah, I was thinking of keeping the JS system that toggles between the form and the success message depending on the query.

Reply:
That's what I was trying to note here, yes. It probably needs an Asana ticket.

I'm going to back away from this ticket since you have some clarity on the approach. Merge at your convenience.